gobblin-compaction/src/main/java/org/apache/gobblin/compaction/action/CompactionCompleteFileOperationAction.java

```diff
@@ -144,28 +147,32 @@ public void onCompactionJobComplete(FileSystemDataset dataset) throws IOExceptio
         Counter counter = job.getCounters().findCounter(RecordKeyMapperBase.EVENT_COUNTER.RECORD_COUNT);
         newTotalRecords = counter.getValue();
       }
+      final Path finalDstPath = dstPath;
       goodPaths.stream().forEach(p -> {
         String fileName = p.getName();
-        outputFiles.add(new Path(dstPath, fileName));
+        outputFiles.add(new Path(finalDstPath, fileName));
       });
       this.configurator.setDstNewFiles(outputFiles);
 
       State compactState = helper.loadState(new Path(result.getDstAbsoluteDir()));
-      if(executeCount!=0) {
-        compactState.setProp(CompactionSlaEventHelper.RECORD_COUNT_TOTAL + Long.toString(executeCount), Long.toString(helper.readRecordCount(new Path(result.getDstAbsoluteDir()))));
-        compactState.setProp(CompactionSlaEventHelper.EXEC_COUNT_TOTAL + Long.toString(executeCount), Long.toString(executeCount));
-        compactState.setProp("DuplicateRecordCount" + Long.toString(executeCount), compactState.getProp("DuplicateRecordCount", "null"));
+      if (executeCount != 0) {
+        compactState.setProp(CompactionSlaEventHelper.RECORD_COUNT_TOTAL + Long.toString(executeCount),
+            Long.toString(helper.readRecordCount(new Path(result.getDstAbsoluteDir()))));
+        compactState.setProp(CompactionSlaEventHelper.EXEC_COUNT_TOTAL + Long.toString(executeCount),
+            Long.toString(executeCount));
+        compactState.setProp("DuplicateRecordCount" + Long.toString(executeCount),
+            compactState.getProp("DuplicateRecordCount", "null"));
       }
       compactState.setProp(CompactionSlaEventHelper.RECORD_COUNT_TOTAL, Long.toString(newTotalRecords));
       compactState.setProp(CompactionSlaEventHelper.EXEC_COUNT_TOTAL, Long.toString(executeCount + 1));
       compactState.setProp(CompactionSlaEventHelper.MR_JOB_ID,
           this.configurator.getConfiguredJob().getJobID().toString());
-      compactState.setProp("DuplicateRecordCount", job.getCounters().findCounter(
-          RecordKeyDedupReducerBase.EVENT_COUNTER.DEDUPED).getValue());
-      compactState.setProp(CompactionSlaEventHelper.LAST_RUN_START_TIME, this.state.getProp(CompactionSource.COMPACTION_INIT_TIME));
+      compactState.setProp("DuplicateRecordCount",
+          job.getCounters().findCounter(RecordKeyDedupReducerBase.EVENT_COUNTER.DEDUPED).getValue());
+      compactState.setProp(CompactionSlaEventHelper.LAST_RUN_START_TIME,
+          this.state.getProp(CompactionSource.COMPACTION_INIT_TIME));
       helper.saveState(new Path(result.getDstAbsoluteDir()), compactState);
-      log.info("duplicated records count for "+ dstPath + " : " + compactState.getProp("DuplicateRecordCount"));
-
+      log.info("duplicated records count for " + dstPath + " : " + compactState.getProp("DuplicateRecordCount"));
```

Comment:
What happens when executeCount = 0? Will compactState.getProp() return a non-null value?

Reply:
Here we already set the compactState to the state of current execution, so there should be valid value there. executionCount calculated from previous run's state will be 0 it it's the first time of compaction